### PR TITLE
prevent out of boundary

### PR DESCRIPTION
**Pull request by vivid392845427**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -288,17 +288,16 @@ func RdColumnsAndPk(leastCol int) ([]ColumnType, []ColumnType) {
 }
 
 func rdColumns(least int) []ColumnType {
-	colCnt = util.RdRange(columnLeast, columnMost)
-	if colCnt < least {
-		colCnt = least
+	colNum := util.RdRange(columnLeast, columnMost)
+	colCnt = colNum
```
Comment (you06) [inline main.go]: IIRC, colCnt is a global variable which will be used when generating DMLs. So the value of colCnt may be equal to the count of columns in a table, I mean putting this value assign after the if section(where colNum is finally determinied).

**Comment by you06** [comment]
Can you add a description of this PR? I don't know where the "out of boundary" error may be occurred.

**Comment by vivid392845427** [comment]
> Can you add a description of this PR? I don't know where the "out of boundary" error may be occurred.

colCnt is a global variable that will be modified by multiple ddl thread. If one thread sets the colCnt value and specifies the number of slices, then another thread changes colCnt greater than the old, the for loop will report an out-of-bounds error

**Comment by you06** [comment]
@vivid392845427 I took a look at the code. When I writing down the first version, `RdColumnsAndPk`(it will call `rdColumns` and set a global variable) is used for prepare the test schema only.

After we added more DDL cases when DMLs are executed, this function will also call `RdColumnsAndPk`, and this function will be called in concurrent goroutines, which will lead to the out of boundary error.

https://github.com/PingCAP-QE/amend-random/blob/26505579705860e598ace4070cf429659d45e95e/ddl.go#L332-L333

So I think there should no global variable, but for the simplicity, you may just add a no-global-variable version of `RdColumnsAndPk` and `rdColumns` or some solution else.

File: main.go [hunk 2]
```diff
@@ -379,6 +379,7 @@ func once(db, db2 *sql.DB, log *Log) error {
 		leastCol = 100
 	}
 	columns, primary := RdColumnsAndPk(leastCol)
+	colCnt = len(columns)
```
Comment (you06) [inline main.go:382]: Great fix!

**Review by you06 — APPROVED** [review]
LGTM

**Pull request merged**